### PR TITLE
Fix threads timeline message ordering

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/21113
Fixes https://github.com/vector-im/element-web/issues/20914

To review alongside https://github.com/matrix-org/matrix-js-sdk/pull/2215

- Fix threads timeline message ordering with msc3440 support
- Fix message ordering when no thread server support
- Reset pagination state when adding events to a thread timeline

---
This PR currently has no changelog labels, so will not be included in changelogs.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

Preview: https://pr7968--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/structures/ThreadView.tsx [hunk 1]
```diff
@@ -241,29 +243,35 @@ export default class ThreadView extends React.Component<IProps, IState> {
...
+
     private onPaginationRequest = async (
         timelineWindow: TimelineWindow | null,
         direction = Direction.Backward,
         limit = 20,
     ): Promise<boolean> => {
         if (!Thread.hasServerSideSupport) {
-            return false;
+            timelineWindow.extend(direction, limit);
+            return true;
```
Comment (t3chguy) [inline src/components/structures/ThreadView.tsx:255]: I wonder if we can switch this back to `return timelineWindow.extend(direction, limit);` now that `Room.timeline` events re-set the canPaginate* states

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Comment by t3chguy** [comment]
PR title is a little misleading, given this also fixes ordering without the msc support (degraded mode) too

File: src/components/structures/ThreadView.tsx (hunk 1 above)
Comment (gsouquet) [reply src/components/structures/ThreadView.tsx:255]: yup, i'll test this out

**Pull request merged**